design/continuous-delivery.md

```diff
@@ -0,0 +1,578 @@
+      "type": "aws:cloudformation:stack",
+      "environment": "aws://111111111DEP/us-east-1",
+      "properties": {
+        "templateFile": "pipeline-main.template.json",
+        "deployRoleArn": "aws-cdk-deploy-111111111DEP-us-east-1",
+        "adminRoleArn": "aws-cdk-admin-111111111DEP-us-east-1"
+      }
+    },
+    "pipeline-eu-west-2": {
+      "type": "aws:cloudformation:stack",
+      "environment": "aws://111111111DEP/eu-west-2",
+      "properties": {
+        "templateFile": "pipeline-main.template.json",
+        "deployRoleArn": "aws-cdk-deploy-111111111DEP-eu-west-2",
+        "adminRoleArn": "aws-cdk-admin-111111111DEP-eu-west-2"
+      }
+    }
+  }
+}
+```
+
+The `assets.json` file will look like this:
+
+```json
+{
+  "version": "assets-1.0",
+  "images": {
+    "d31ca1aef8d1b68217852e7aea70b1e857d107b47637d5160f9f9a1b24882d2a": {
+      "source": {
+        "packaging": "docker",
+        "sourceHash": "d31ca1aef8d1b68217852e7aea70b1e857d107b47637d5160f9f9a1b24882d2a",
+        "sourcePath": "my-image",
+        "dockerfile": "CustomDockerFile"
+      },
+      "destinations": [
+        {
+          "repositoryName": "aws-cdk-images-2222222222US-us-east-1",
+          "imageName": "d31ca1aef8d1b68217852e7aea70b1e857d107b47637d5160f9f9a1b24882d2a",
+          "assumeRoleArn": "arn:aws:iam::2222222222US:role/aws-cdk-publish-2222222222US-us-east-1"
+        },
+        {
+          "repositoryName": "aws-cdk-images-3333333333EU-eu-west-2",
+          "imageName": "d31ca1aef8d1b68217852e7aea70b1e857d107b47637d5160f9f9a1b24882d2a",
+          "assumeRoleArn": "arn:aws:iam::3333333333EU:role/aws-cdk-publish-3333333333EU-eu-west-2"
+        }
+      ]
+    }
+  },
+  "files": {
+    "a0bae29e7b47044a66819606c65d26a92b1e844f4b3124a5539efc0167a09e57": {
+      "source": {
+        "packaging": "zip",
+        "sourceHash": "a0bae29e7b47044a66819606c65d26a92b1e844f4b3124a5539efc0167a09e57",
+        "sourcePath": "asset.a0bae29e7b47044a66819606c65d26a92b1e844f4b3124a5539efc0167a09e57"
+      },
+      "destinations": [
+        {
+          "bucketName": "aws-cdk-files-2222222222US-us-east-1",
+          "objectKey": "a0bae29e7b47044a66819606c65d26a92b1e844f4b3124a5539efc0167a09e57",
+          "assumeRoleArn": "arn:aws:iam::2222222222US:role/aws-cdk-publish-2222222222US-us-east-1"
+        },
+        {
+          "bucketName": "aws-cdk-files-3333333333EU-us-west-2",
+          "objectKey": "a0bae29e7b47044a66819606c65d26a92b1e844f4b3124a5539efc0167a09e57",
+          "assumeRoleArn": "arn:aws:iam::3333333333EU:role/aws-cdk-publish-3333333333EU-eu-west-2"
+        }
+      ]
+    }
+  }
+}
+```
+
+It lists the two assets (one file asset and one image asset) and, for each, it lists the publishing locations which include repository, key and publishing IAM role to assume.
+
+### Mutation
+
+The first stage in our pipeline is the mutation stage. This stage will include a single CodeBuild action that will simply execute:
```

Comment:
Before this, call out what the customer experience would be when there's no pipeline yet, i.e., starting out from scratch.

Reply:
added a "pipeline initialization" step